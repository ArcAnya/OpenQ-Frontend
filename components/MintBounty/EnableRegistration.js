import React from 'react';
import ToolTipNew from '../Utils/ToolTipNew';

const EnableRegistration = ({ enableRegistrationState, registrationDeadlineState, startDateState, category }) => {
  const [enableRegistration, setEnableRegistration] = enableRegistrationState;
  const [registrationDeadline, setRegistrationDeadline] = registrationDeadlineState;
  const [startDate, setStartDate] = startDateState;
  return (
    <>
      <div className=' flex flex-col gap-2 w-full py-2 items-start text-base bg-[#161B22]'>
        <div className='flex items-center gap-2 font-semibold'>
          Enable Hackathon Registration
          <input type='checkbox' className='checkbox' onChange={() => setEnableRegistration(true)}></input>
          <ToolTipNew
            mobileX={10}
            toolTipText={
              category === 'Fixed Price'
                ? 'Amount of funds you would like to escrow on this issue.'
                : 'How much will each successful submitter earn?'
            }
          >
            <div className='cursor-help rounded-full border border-[#c9d1d9] aspect-square leading-4 h-4 box-content text-center font-bold text-primary'>
              ?
            </div>
          </ToolTipNew>
        </div>
        <span className='text-sm '>
          Require contestants to sign up for your hackathon contests in this repo. This will allow you to set a
          timeline, be highlighted on OpenQ, and ensure you can connect with all participants post-hackathon.
        </span>
      </div>
      {enableRegistration ? (
        <>
          <div className='flex items-center gap-2'>Start Date</div>

          <input
            className={'flex-1 input-field w-full ml-2'}
            id='name'
            aria-label='issue url'
            placeholder='https://github.com/...'
            autoComplete='off'
            type='date'
            value={startDate}
            onChange={(e) => setStartDate(e.target.value)}
          />
          <div className='flex items-center gap-2'>End Date</div>

          <input
            className={'flex-1 input-field w-full ml-2'}
            id='name'
            aria-label='issue url'
            placeholder='https://github.com/...'
            autoComplete='off'
            type='date'
            value={registrationDeadline}
            onChange={(e) => setRegistrationDeadline(e.target.value)}
          />
        </>
      ) : null}
    </>
  );
};
export default EnableRegistration;
