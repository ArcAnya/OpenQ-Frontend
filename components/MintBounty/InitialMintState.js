const zeroAddressMetadata = {
  name: 'Matic',
  address: '0x0000000000000000000000000000000000000000',
  symbol: 'MATIC',
  decimals: 18,
  chainId: 80001,
  path: 'https://wallet-asset.matic.network/img/tokens/matic.svg',
};

const InitialMintState = {
  finalTierVolumes: [1, 1, 1],
  isLoading: false,
  payoutVolume: '',
  payoutToken: zeroAddressMetadata,
  enableRegistration: false,
  registrationDeadline: null,
  startDate: null,
  sum: 0,
  hideModal: false,
  invoiceable: false,
  goalVolume: '',
  goalToken: zeroAddressMetadata,
};

export default InitialMintState;
