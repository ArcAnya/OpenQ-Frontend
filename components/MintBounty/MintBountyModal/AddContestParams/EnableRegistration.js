import React, { useContext } from 'react';
import ToolTipNew from '../../../Utils/ToolTipNew';
import MintContext from '../../MintContext';

const EnableRegistration = () => {
  const [mintState, mintDispatch] = useContext(MintContext);
  const { startDate, registrationDeadline, enableRegistration, category } = mintState;
  const handleStartDate = (e) => {
    const dispatch = {
      type: 'UPDATE_START_DATE',
      payload: e.target.value,
    };
    mintDispatch(dispatch);
  };

  const handleRegistrationDeadline = (e) => {
    const dispatch = {
      type: 'UPDATE_REGISTRATION_DEADLINE',
      payload: e.target.value,
    };
    mintDispatch(dispatch);
  };
  const handleRegistration = (e) => {
    const dispatch = {
      type: 'UPDATE_ENABLE_REGISTRATION',
      payload: e.target.checked,
    };
    mintDispatch(dispatch);
  };

  return (
    <>
      <div className=' flex flex-col gap-2 w-full py-2 items-start text-base bg-[#161B22]'>
        <div className='flex items-center gap-2 font-semibold'>
          <label htmlFor='enable registration checkbox'>Enable Hackathon Registration</label>
          <input
            type='checkbox'
            className='checkbox'
            id='enable registration checkbox'
            checked={enableRegistration}
            onChange={handleRegistration}
          ></input>
          <ToolTipNew
            mobileX={10}
            toolTipText={
              category === 'Fixed Price'
                ? 'Amount of funds you would like to escrow on this issue.'
                : 'How much will each successful submitter earn?'
            }
          >
            <div className='cursor-help rounded-full border border-[#c9d1d9] aspect-square leading-4 h-4 box-content text-center font-bold text-primary'>
              ?
            </div>
          </ToolTipNew>
        </div>
        <span className='text-sm '>
          Require contestants to sign up for your hackathon contests in this repo. This will allow you to set a
          timeline, be highlighted on OpenQ, and ensure you can connect with all participants post-hackathon.
        </span>
      </div>
      {enableRegistration ? (
        <>
          <label htmlFor='start date' className='flex items-center gap-2'>
            Start Date
          </label>
          <input
            className={'flex-1 input-field w-full'}
            id='start date'
            aria-label='Start Date'
            data-testid='start date'
            placeholder='https://github.com/...'
            autoComplete='off'
            type='date'
            value={startDate}
            onChange={handleStartDate}
          />
          <label htmlFor='end date' className='flex items-center gap-2'>
            End Date
          </label>

          <input
            className={'flex-1 input-field w-full'}
            id='end date'
            placeholder='https://github.com/...'
            autoComplete='off'
            type='date'
            value={registrationDeadline}
            onChange={handleRegistrationDeadline}
          />
        </>
      ) : null}
    </>
  );
};
export default EnableRegistration;
