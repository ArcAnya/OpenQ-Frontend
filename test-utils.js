// Third party
import React from 'react';
// Cusotm
import { render } from '@testing-library/react';
import StoreProvider from './store/Store/StoreProvider';
import AuthProvider from './store/AuthStore/AuthProvider';

// Add in any providers here if necessary:
// test-utils.js
// (ReduxProvider, ThemeProvider, etc)
const Providers = ({ children }) => {
  return (
    <AuthProvider>
      <StoreProvider>{children}</StoreProvider>
    </AuthProvider>
  );
};

const customRender = (ui, options = {}) => render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
