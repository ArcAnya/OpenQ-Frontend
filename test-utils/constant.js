export default class Constants {
  constructor() {}
  static get githubUserId() {
    return 'MDQ6VXNlcjcyMTU2Njc5';
  }

  static get uuid() {
    return '63a3a266ab24d99d359a6d51';
  }

  static get email() {
    return 'abc123@gmail.com';
  }

  static get userName() {
    return 'Christopher';
  }
  static getTwitter() {
    return 'https://twitter.com/Christo28120856';
  }
  static getWatchedBounties() {
    return { __typename: 'Bounties', nodes: [] };
  }
  static getStarredOrganizationIds() {
    return [];
  }

  static get accountData() {
    return {
      id: Constants.uuid,
      github: Constants.githubUserId,
      email: Constants.email,
      username: Constants.userName,
      twitter: 'https://twitter.com/Christo28120856',

      watchedBounties: Constants.watchedBounties,
      starredOrganizationIds: Constants.starredOrganizationIds,
    };
  }

  static get bountyId() {
    return 'I_kwDOGAqhQc48U5_r';
  }
  static get claimant() {
    return { id: '0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266', __typename: 'User' };
  }
  static get claimantAsset() {
    return 'https://github.com/ArcAnya/OpenQ-TestRepo/pull/153';
  }
  static get claim() {
    return {
      claimTime: Constants.claimTime,
      claimant: Constants.claimant,
      claimantAsset: Constants.claimantAsset,
      __typename: 'Claim',
    };
  }
  static get claimTime() {
    return '1661767948472';
  }
  static get body() {
    return 'body of test2';
  }
  static get url() {
    return 'https://github.com/OpenQDev/OpenQ-TestRepo/issues/8';
  }
  static get languages() {
    return [Constants.language1, Constants.language2, Constants.language3];
  }
  static get repoName() {
    return 'OpenQ-Frontend';
  }
  static get owner() {
    return 'OpenQDev';
  }
  static get avatarUrl() {
    return 'https://avatars.githubusercontent.com/u/77402538?v=4';
  }
  static get labels() {
    return [Constants.label1, Constants.label2];
  }
  static get createdAt() {
    return '1661767948472';
  }
  static get closed() {
    return true;
  }
  static get bodyHTML() {
    return '<p dir="auto">body of test2</p>';
  }
  static get title() {
    return 'Properly Referenced and Merged by FlacoJones';
  }
  static get titleHTML() {
    return 'Properly Referenced and Merged by FlacoJones';
  }
  static get tvl() {
    return '9.51848';
  }

  static get assignees() {
    return [];
  }
  static get twitterUsername() {
    return 'openqlabs';
  }
  static get number() {
    return 8;
  }

  static get authState() {
    return {
      login: 'Christopher-Stevers',
    };
  }
  static get prs() {
    return [Constants.pr1];
  }
  static get pr1() {
    return {
      __typename: 'CrossReferencedEvent',
      id: 'CRE_kwDOGWnnz85GHUw1',
      assignees: Constants.assignees,
      referencedAt: '2022-03-28T17:47:26Z',
      source: {
        __typename: 'PullRequest',
        mergedAt: '2022-03-28T17:57:44Z',
        url: 'https://github.com/OpenQDev/OpenQ-TestRepo/pull/138',
        id: 'PR_kwDOGWnnz841LGsK',
        merged: true,
        bodyText: 'This Closes #136 and also unrelated thing of Fixes #137',
        body: 'This Closes #136 and also unrelated thing of Fixes #137',
        bodyHTML:
          '<p dir="auto">This <span class="issue-keyword tooltipped tooltipped-se" aria-label="This pull request closes issue #136.">Closes</span> <a class="issue-link js-issue-link" data-error-text="Failed to load title" data-id="1183776821" data-permission-text="Title is private" data-url="https://github.com/OpenQDev/OpenQ-TestRepo/issues/136" data-hovercard-type="issue" data-hovercard-url="/OpenQDev/OpenQ-TestRepo/issues/136/hovercard" href="https://github.com/OpenQDev/OpenQ-TestRepo/issues/136">#136</a> and also unrelated thing of <span class="issue-keyword tooltipped tooltipped-se" aria-label="This pull request closes issue #137.">Fixes</span> <a class="issue-link js-issue-link" data-error-text="Failed to load title" data-id="1183777365" data-permission-text="Title is private" data-url="https://github.com/OpenQDev/OpenQ-TestRepo/issues/137" data-hovercard-type="issue" data-hovercard-url="/OpenQDev/OpenQ-TestRepo/issues/137/hovercard" href="https://github.com/OpenQDev/OpenQ-TestRepo/issues/137">#137</a></p>',
        title: 'Update README.md',
        author: {
          __typename: 'User',
          login: 'FlacoJones',
          user: { __typename: 'User', login: 'FlacoJones', url: 'https://github.com/FlacoJones' },

          avatarUrl: 'https://avatars.githubusercontent.com/u/93455288?u=fd1fb04b6ff2bf397f8353eafffc3bfb4bd66e84&v=4',
          url: 'https://github.com/FlacoJones',
          id: 'U_kgDOBZIDuA',
          email: '',
        },
        mergeCommit: {
          __typename: 'Commit',
          author: {
            __typename: 'GitActor',
            avatarUrl: 'https://avatars.githubusercontent.com/u/93455288?v=4',
            name: 'FlacoJones',
            user: { __typename: 'User', login: 'FlacoJones', url: 'https://github.com/FlacoJones' },
          },
        },
      },
    };
  }

  static get closedEvents() {
    return [Constants.closedEvent1];
  }
  static get closedAt() {
    return '2022-03-28T17:57:44Z';
  }
  static get language1() {
    return {
      __typename: 'Language',
      name: 'JavaScript',
      color: '#f1e05a',
    };
  }
  static get language2() {
    return {
      __typename: 'Language',
      name: 'CSS',
      color: '#563d7c',
    };
  }
  static get language3() {
    return {
      __typename: 'Language',
      name: 'Dockerfile',
      color: '#384d54',
    };
  }
  static get language4() {
    return {
      __typename: 'Language',
      name: 'Shell',
      color: '#89e051',
    };
  }
  static get label1() {
    return {
      __typename: 'Label',
      name: 'documentation',
      color: '0075ca',
    };
  }
  static get label2() {
    return {
      __typename: 'Label',
      name: 'duplicate',
      color: 'cfd3d7',
    };
  }
  static get label3() {
    return {
      __typename: 'Label',
      name: 'enhancement',
      color: 'a2eeef',
    };
  }

  static get label4() {
    return {
      __typename: 'Label',
      name: 'javascript',
      color: 'E16391',
    };
  }
  static get label5() {
    return {
      __typename: 'Label',
      name: 'bounty',
      color: '727384',
    };
  }
  static get repoUrl() {
    return 'https://github.com/OpenQDev/OpenQ-Frontend';
  }
  static get repoDescription() {
    return null;
  }
  static get closedEvent1() {
    return {
      __typename: 'ClosedEvent',
      id: 'CE_lADOGWnnz85GjwA1zwAAAAF4vHFc',
      actor: {
        __typename: 'User',
        avatarUrl: 'https://avatars.githubusercontent.com/u/93455288?u=fd1fb04b6ff2bf397f8353eafffc3bfb4bd66e84&v=4',
        login: 'FlacoJones',
        url: 'https://github.com/FlacoJones',
        name: null,
      },
    };
  }

  static get bountyAddress() {
    return '0x3c57cd59335d9a56b76feb48b3925d095e7e8e9f';
  }
  static get bountyMintTime() {
    return '1662545344';
  }
  static get bountyClosedTime() {
    return null;
  }
  static get status() {
    return '0';
  }
  static get fundingGoalTokenAddress() {
    return '0x0000000000000000000000000000000000000000';
  }
  static get fundingGoalVolume() {
    return '0';
  }
  static get payoutTokenVolume() {
    return '100';
  }
  static get payoutTokenAddress() {
    return '0x5fbdb2315678afecb367f032d93f642f64180aa3';
  }
  static get payoutSchedule0() {
    return null;
  }
  static get payoutSchedule2() {
    return ['30', '60', '10'];
  }
  static get closerData() {
    return null;
  }
  static get bountyType() {
    return '1';
  }
  static get claimedTransactionHash() {
    return null;
  }
  static get deposits() {
    return [Constants.deposit1, Constants.deposit2, Constants.deposit3];
  }

  static get deposit1() {
    return {
      id: '0xe5551a3fa87d93a0c6c084d572b9e282114befc43dc68f08be6d53d13830e356',
      refunded: false,
      receiveTime: '1662545374',
      tokenAddress: '0x0000000000000000000000000000000000000000',
      expiration: '1296000',
      volume: '1000000000000000000',
      refundTime: null,
      sender: { id: '0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266', __typename: 'User' },
      __typename: 'Deposit',
    };
  }
  static get deposit2() {
    return {
      id: '0xb4f31aab8a1c4bfe26236729e8cd8e4abf81d63283e006b4ec677a7ce6b2871a',
      refunded: true,
      receiveTime: '1662545373',
      tokenAddress: '0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0',
      expiration: '30',
      volume: '2000000000000000000',
      refundTime: '1662559726',
      sender: { id: '0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266', __typename: 'User' },
      __typename: 'Deposit',
    };
  }
  static get deposit3() {
    return {
      id: '0x7db2691e573b9c19e90f391cd3eda9ce9246666a1502f2bf87b9d47d13679bc0',
      refunded: false,
      receiveTime: '1662545372',
      tokenAddress: '0x5fbdb2315678afecb367f032d93f642f64180aa3',
      expiration: '1296000',
      volume: '1000000000000000000',
      refundTime: null,
      sender: { id: '0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266', __typename: 'User' },
      __typename: 'Deposit',
    };
  }
  static get payouts() {
    return [Constants.payout1, Constants.payout2, Constants.payout3];
  }

  static get payout1() {
    return {
      tokenAddress: Constants.maticAddress,
      volume: '700000000000000000000',
      closer: { id: '0x3c44cdddb6a900fa2b585dd299e03d12fa4293bc', __typename: 'User' },
      __typename: 'Payout',
    };
  }

  static get payout2() {
    return {
      tokenAddress: Constants.linkAddress,
      volume: '300000000000000000000',
      payoutTime: '1663576393',
      closer: { id: '0x3c44cdddb6a900fa2b585dd299e03d12fa4293bc', __typename: 'User' },
      __typename: 'Payout',
    };
  }

  static get payout3() {
    return {
      tokenAddress: Constants.daiAddress,
      volume: Constants.volume,
      payoutTime: '1663576393',
      closer: { id: '0x3c44cdddb6a900fa2b585dd299e03d12fa4293bc', __typename: 'User' },
      __typename: 'Payout',
    };
  }
  static get refunds() {
    return [Constants.refund1];
  }
  static get refund1() {
    return {
      refundTime: '1662559726',
      tokenAddress: Constants.daiAddress,
      volume: Constants.volume,
      depositId: '0xb4f31aab8a1c4bfe26236729e8cd8e4abf81d63283e006b4ec677a7ce6b2871a',
      __typename: 'Refund',
    };
  }

  static get issuer() {
    return {
      __typename: 'User',
      id: '0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266',
    };
  }
  static get bountyTokenBalances() {
    return [Constants.bountyTokenBalanceMatic, Constants.bountyTokenBalance1, Constants.bountyTokenBalance1];
  }
  static get volume() {
    return '2000000000000000000';
  }
  static get bountyTokenBalanceMatic() {
    return {
      __typename: 'BountyFundedTokenBalance',
      volume: Constants.volume,
      tokenAddress: Constants.maticAddress,
    };
  }
  static get bountyTokenBalanceDai() {
    return {
      __typename: 'BountyFundedTokenBalance',
      volume: Constants.volume,
      tokenAddress: Constants.daiAddress,
    };
  }
  static get bountyTokenBalanceLink() {
    return {
      __typename: 'BountyFundedTokenBalance',
      volume: Constants.volume,
      tokenAddress: Constants.linkAddress,
    };
  }
  static get linkAddress() {
    return '0x5FbDB2315678afecb367f032d93F642f64180aa3';
  }
  static get daiAddress() {
    return '0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0';
  }
  static get maticAddress() {
    return '0x0000000000000000000000000000000000000000';
  }
  static get address() {
    return '0x3c57cd59335d9a56b76feb48b3925d095e7e8e9f';
  }
  static get organizationId() {
    return '0x3c57cd59335d9a56b76feb48b3925d095e7e8e9f';
  }
  static get type() {
    return '1';
  }
  static get category() {
    return '1';
  }
  static get watchingCount() {
    return 0;
  }
  static get blacklisted() {
    return false;
  }
  static get organization() {
    return { blacklisted: false, __typename: 'Organization' };
  }
  static get bountyTypeFixed() {
    return '0';
  }
  static get bountyTypeSplit() {
    return '1';
  }
  static get bountyTypeContest() {
    return '2';
  }
  static get bountyTypeFixedContest() {
    return '3';
  }
  static get bounty() {
    return {
      id: Constants.id,
      title: Constants.title,
      assignees: Constants.assignees,
      body: Constants.body,
      url: Constants.url,
      repoName: Constants.repoName,
      closedAt: Constants.closedAt,
      owner: Constants.owner,
      avatarUrl: Constants.avatarUrl,
      labels: Constants.labels,
      createdAt: Constants.createdAt,
      closed: Constants.closed,
      bodyHTML: Constants.bodyHTML,
      titleHTML: Constants.titleHTML,
      twitterUsername: Constants.twitterUsername,
      number: Constants.number,
      prs: Constants.prs,
      closedEvents: Constants.closedEvents,
      tvl: Constants.tvl,
      bountyId: Constants.bountyId,
      bountyAddress: Constants.bountyAddress,
      closerData: Constants.closerData,
      bountyMintTime: Constants.bountyMintTime,
      deposits: Constants.deposits,
      refunds: Constants.refunds,
      bountyTokenBalances: Constants.bountyTokenBalances,
      issuer: Constants.issuer,
      bountyType: Constants.bountyTypeFixed,
      payoutSchedule: Constants.payoutSchedule0,
    };
  }
}
