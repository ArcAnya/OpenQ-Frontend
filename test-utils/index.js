// Third party
import React from 'react';
// Cusotm
import { render, renderHook } from '@testing-library/react';

import StoreProvider from '../store/Store/TestStoreProvider';
import AuthProvider from '../store/AuthStore/TestAuthProvider';
import FundProvider from '../components/FundBounty/TestFundProvider';

// Add in any providers here if necessary:
// test-utils.js
// (ReduxProvider, ThemeProvider, etc)
const customRender = (ui, options = {}, storeProps, authProps) => {
  const Providers = ({ children }) => {
    return (
      <AuthProvider StoreProps={authProps}>
        <StoreProvider StoreProps={storeProps}>{children}</StoreProvider>
      </AuthProvider>
    );
  };

  render(ui, { wrapper: Providers, ...options });
};
const customHookRender = (hook, options = {}, storeProps, authProps, fundProps) => {
  const Providers = ({ children }) => {
    return (
      <AuthProvider StoreProps={authProps}>
        <StoreProvider StoreProps={storeProps}>
          <FundProvider StoreProps={fundProps}> {children}</FundProvider>
        </StoreProvider>
      </AuthProvider>
    );
  };

  return renderHook(hook, { wrapper: Providers, ...options });
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render, customHookRender as hookRender };
