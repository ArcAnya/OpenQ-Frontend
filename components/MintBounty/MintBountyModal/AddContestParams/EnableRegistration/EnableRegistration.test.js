/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, screen } from '../../../../../test-utils';
import EnableRegistration from '../../../MintBountyModal/AddContestParams/EnableRegistration';
import InitialMintState from '../../../InitialMintState';
import MintContext from '../../../MintContext';
import userEvent from '@testing-library/user-event';

//const mockTierVolumeChange = jest.fn();
describe('Enable Registration', () => {
  const currentDate = new Date('01-01-2023');
  it('should fire event', async () => {
    // ARRANGE
    const mintState = {
      ...InitialMintState.mintState,
      startDate: currentDate,
      enableRegistration: true,
    };
    const mintDispatch = jest.fn();

    const user = await userEvent.setup();
    render(
      <MintContext.Provider value={[mintState, mintDispatch]}>
        <EnableRegistration />
      </MintContext.Provider>
    );

    // ACT
    await user.click(screen.getByLabelText(/Enable Hackathon Registration/));

    // ASSERT
    expect(mintDispatch).toHaveBeenCalledWith({
      type: 'UPDATE_ENABLE_REGISTRATION',
      payload: false,
    });
    // expect startdate to be in input
  });

  it('should display inputs', async () => {
    // ARRANGE
    const mintState = {
      ...InitialMintState.mintState,
      startDate: currentDate,
      enableRegistration: true,
    };
    const mintDispatch = jest.fn();

    const { asFragment } = render(
      <MintContext.Provider value={[mintState, mintDispatch]}>
        <EnableRegistration />
      </MintContext.Provider>
    );

    // ASSERT
    expect(asFragment()).toMatchSnapshot();
  });
});
