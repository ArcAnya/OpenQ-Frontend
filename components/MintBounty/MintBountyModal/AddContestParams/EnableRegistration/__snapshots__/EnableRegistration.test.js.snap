// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enable Registration should display inputs 1`] = `
<DocumentFragment>
  <div
    class=" flex flex-col gap-2 w-full py-2 items-start text-base bg-[#161B22]"
  >
    <div
      class="flex items-center gap-2 font-semibold"
    >
      <label
        for="enable registration checkbox"
      >
        Enable Hackathon Registration
      </label>
      <input
        checked=""
        class="checkbox"
        id="enable registration checkbox"
        type="checkbox"
      />
    </div>
    <span
      class="text-sm "
    >
      Require contestants to sign up for your hackathon contests in this repo. This will allow you to set a timeline, be highlighted on OpenQ, and ensure you can connect with all participants post-hackathon.
    </span>
  </div>
  <label
    class="flex items-center gap-2"
    for="start date"
  >
    Start Date
  </label>
  <input
    aria-label="Start Date"
    autocomplete="off"
    class="flex-1 input-field w-full"
    data-testid="start date"
    id="start date"
    max="2100-01-01"
    min="2023-01-01"
    placeholder="https://github.com/..."
    type="date"
    value="Sun Jan 01 2023 00:00:00 GMT+0000 (Coordinated Universal Time)"
  />
  <label
    class="flex items-center gap-2"
    for="end date"
  >
    End Date
  </label>
  <input
    autocomplete="off"
    class="flex-1 input-field w-full"
    id="end date"
    max="2100-01-01"
    min="2023-01-01"
    placeholder="https://github.com/..."
    type="date"
    value=""
  />
</DocumentFragment>
`;
